lib/ClangImporter/ImportType.cpp

```diff
@@ -1986,9 +1979,11 @@ Type ClangImporter::Implementation::importMethodType(
         swiftResultTy = OptionalType::get(OptionalityOfReturn, swiftResultTy);
     }
   }
+
   if (!swiftResultTy)
     return Type();
-  swiftResultTy = mapTypeIntoContext(swiftResultTy);
+
+  swiftResultTy = mapTypeIntoContext(origDC, dc, swiftResultTy);
```

Comment:
Okay, I understand this now. The contexts can be different in cases where (for example) we're importing an init method from a superclass into an initializer in a subclass. Sorry for the noise.